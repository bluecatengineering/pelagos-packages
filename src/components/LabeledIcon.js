import PropTypes from 'prop-types';

import SvgIcon from './SvgIcon';
import './LabeledIcon.less';

/** An icon with a label underneath. */
const LabeledIcon = ({className, icon, label, underline, ...props}) => (
	<div {...props} className={'LabeledIcon' + (className ? ` ${className}` : '')} role="button">
		<SvgIcon className="LabeledIcon__icon" icon={icon} />
		<span className={'LabeledIcon__label' + (underline ? ' LabeledIcon--underlined' : '')}>{label}</span>
	</div>
);

LabeledIcon.propTypes = {
	/** The component class name(s). */
	className: PropTypes.string,
	/** The icon to display. */
	icon: PropTypes.object,
	/** The label text. */
	label: PropTypes.string,
	/** Whether the first letter of the label is underlined. */
	underline: PropTypes.bool,
};

export default LabeledIcon;
