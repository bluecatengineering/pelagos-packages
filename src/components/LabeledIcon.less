@import '../../less/fonts';
@import '../../less/spacing';
@import '../../less/utils';

.LabeledIcon {
	@icon-button();
	align-items: center;
	padding: @sp-04;
}

.LabeledIcon__icon {
	font-size: 20px;
}

.LabeledIcon__label {
	@font-10-600-uc();
	margin-top: @sp-02;
	white-space: nowrap;
}

.LabeledIcon--underlined::first-letter {
	text-decoration: underline;
}
