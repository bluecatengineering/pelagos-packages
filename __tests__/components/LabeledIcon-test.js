import {shallow} from 'enzyme';

import LabeledIcon from '../../src/components/LabeledIcon';

jest.unmock('../../src/components/LabeledIcon');

describe('LabeledIcon', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<LabeledIcon icon={{icon: 'test'}} label="test" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when className is set', () => {
			const wrapper = shallow(<LabeledIcon icon={{icon: 'test'}} className="Test" label="test" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when underline is set', () => {
			const wrapper = shallow(<LabeledIcon icon={{icon: 'test'}} label="test" underline />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
