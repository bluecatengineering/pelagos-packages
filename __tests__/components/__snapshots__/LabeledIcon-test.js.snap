// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledIcon rendering renders expected elements 1`] = `
<div
  className="LabeledIcon"
  role="button"
>
  <SvgIcon
    className="LabeledIcon__icon"
    icon={
      Object {
        "icon": "test",
      }
    }
  />
  <span
    className="LabeledIcon__label"
  >
    test
  </span>
</div>
`;

exports[`LabeledIcon rendering renders expected elements when className is set 1`] = `
<div
  className="LabeledIcon Test"
  role="button"
>
  <SvgIcon
    className="LabeledIcon__icon"
    icon={
      Object {
        "icon": "test",
      }
    }
  />
  <span
    className="LabeledIcon__label"
  >
    test
  </span>
</div>
`;

exports[`LabeledIcon rendering renders expected elements when underline is set 1`] = `
<div
  className="LabeledIcon"
  role="button"
>
  <SvgIcon
    className="LabeledIcon__icon"
    icon={
      Object {
        "icon": "test",
      }
    }
  />
  <span
    className="LabeledIcon__label LabeledIcon--underlined"
  >
    test
  </span>
</div>
`;
